### ISSUE TITLE
[BUG]: If `/` is not found, an empty name is returned

### ISSUE THREAD

**Issue by sounisi5011** [mheob/used-pm#27]
https://github.com/mheob/used-pm/blob/bf32d74b22a3a30f49750d9c9959c1ba0ac88a4d/src/index.cts#L14

https://github.com/mheob/used-pm/blob/bf32d74b22a3a30f49750d9c9959c1ba0ac88a4d/src/index.mts#L14

I assume that these codes are probably intended to "return 0 if no slash is found in the variable `packageManager`".
However, [the `indexOf()` method](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/indexOf) returns `-1` if the passed string is not found.
So I think the correct code *for this line* is like this:

```ts
const separatorPosition = Math.max(packageManager.indexOf('/'), 0);
```

However, if "`separatorPosition` is 0 if no slash is found", then by this code following the `separatorPosition` variable:

https://github.com/mheob/used-pm/blob/bf32d74b22a3a30f49750d9c9959c1ba0ac88a4d/src/index.mts#L16-L19

The return value of the `getCurrentPackageManager()` function will look like this:

```js
// packageManager = 'foo'
// separatorPosition = 0
{ name: '', version: 'foo' }
```

...Maybe this return value is also wrong.
In other words, I don't think that fixing the bug in the `separatorPosition` variable will return the correct result.

I don't know the specs of the used-pm, so I can't write anything more about it.

**Comment by mheob** [comment]
You are right. 

```ts
const separatorPosition = packageManager.indexOf('/') ?? 0;
```

… this makes no sense 😁

Currently, only the "default" package managers `npm`, `yarn` and `pnpm` are supported by me. Now that I think about it, I should add this to the README.

Since the script gets the package manager from `process.env.npm_config_user_agent`, I currently assume that there is a `/` in there every time and your example (`{ name: '', version: 'foo' }`) is not possible. Do you have any other examples I should be aware of?

But I also think it would make sense to return a name and an empty version if no `/` was found. I will fix that as well.

**Comment by sounisi5011** [comment]
> Since the script gets the package manager from `process.env.npm_config_user_agent`, I currently assume that there is a `/` in there every time and your example `({ name: '', version: 'foo' })` is not possible.

I think you have the right idea.

I am trying the new package manager [Bun](https://bun.sh/) in addition to the well-known package managers in https://github.com/sounisi5011/nodejs-package-managers-debug--postinstall-script, and [Bun also always includes a `/` in `npm_config_user_agent`](https://github.com/oven-sh/bun/blob/48997eaeba3bb31900b6a688ffcd6bf275511d42/src/cli/run_command.zig#L600-L609) (e.g. `bun/0.5.9 npm/? node/v18.15.0 linux x64`).
In the future, if a new package manager did not include a `/` in `npm_config_user_agent`, it is probably a bug in that package manager.

> Do you have any other examples I should be aware of?

I think there are no other bugs.

<details><summary>If I had to choose, it would be to be able to slightly improve the speed in one location.</summary>

https://github.com/mheob/used-pm/blob/bf32d74b22a3a30f49750d9c9959c1ba0ac88a4d/src/index.mts#L11

[`split()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/split) always finds all ` ` and creates an array. This slows down the speed a bit. Here [`indexOf()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/indexOf) and [`slice()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/slice) can be used to speed up the process.

```ts
const userAgentSeparatorPosition = userAgent.indexOf(' ');
const packageManager = (userAgentSeparatorPosition < 0)
	? userAgent
	: userAgent.slice(0, userAgentSeparatorPosition);
```

However, this effect will only occur if used-pm is run more than a few thousand times.
used-pm will probably not be used for such applications (e.g. computer games).
In this case, it would be better to keep the current code, prioritizing readability over speed.

> **Note**
> A `limit ` argument can be passed to [`split()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/split).
> This is used to limit the length of the returned array.
>
> ```ts
> const packageManager = userAgent.split(' ', 1)[0]; 
> ```
>
> This looks like a good idea to speed up without sacrificing readability.
> However, from my tests with Node.js 18.15.0, passing a `limit` argument **slows it down**.
>
> <details><summary>Here's the code I used for the measurements</summary>
>
> ```js
> const { performance } = require('perf_hooks');
>
> function benchmark(name, testFn) {
>   const start = performance.now();
>   for (let i = 0; i < 9999; i++) {
>     testFn();
>   }
>   const end = performance.now();
>   console.log(name, (end - start) + 'ms');
> }
>
> const userAgent='npm/6.14.18 node/v18.16.0 linux x64 ci/github-actions';
>
> // This is run first to avoid the effect of the dynamic optimization process in JS runtime
> benchmark('do nothing', () => userAgent)
>
> benchmark(`userAgent.split(' ')[0]`,    () => userAgent.split(' ')[0]);
> benchmark(`userAgent.split(' ', 1)[0]`, () => userAgent.split(' ', 1)[0]);
> ```
>
> </details>

</details>

**Comment by mheob** [comment]
> In this case, it would be better to keep the current code, prioritizing readability over speed.

I think you are right here. I will leave it as it is, for better readability. The performance is not the highest priority in this case.
